### ISSUE TITLE
Preprocess script for RAW files

### ISSUE THREAD

**Issue by wangchust** [Linwei-Chen/LIS#7]
The LIS is an excellent work and has contributed a lot to the community. I am new to this domain and have some questions.
1. Is the raw image data used in LIS the same as that in LOD?
2. Can the authors provide the processing script for converting the raw image data into .png files provided in this repo.  I am quite interested in the influence of bit-depth.

**Comment by Linwei-Chen** [comment]
Thank you for your interest! The images in the LIS dataset are essentially the same as those in our previous work, LOD. The processing script is very simple: it first quantifies the raw image and then processes it into PNG format.

**Comment by wangchust** [comment]
Thank you for your reply! I am still a little confused about the raw format. 

In other datasets like SID, raw images are in RGGB bayer format within one channel. I was using rawpy lib to read the raw file and demosaic it into RGB channels using cv2.cvtColor_BayerRGGB2RGB function. Then the so-called RAW-RGB can be quantified and saved as PNG format.

In LIS, do you also follow such pipelines?

**Comment by Linwei-Chen** [comment]
Yes, the raw images are also in RGGB Bayer format within one channel. Please note that, following SID, the G values are averaged from G1 and G2.

**Comment by wangchust** [comment]
Thank you! That's indeed helpful.

**Comment by wangchust** [comment]
> Thank you for your interest! The images in the LIS dataset are essentially the same as those in our previous work, LOD. The processing script is very simple: it first quantifies the raw image and then processes it into PNG format.

By the way, about the quantifying process, can the authors provide the exact code or algorithm?

**Comment by LeonSakura** [comment]
您好，我也十分关注bit-depth 的影响，能否给出原始图像数据转换为此存储库中提供的.png文件的处理脚本，十分感谢！

**Comment by Linwei-Chen** [comment]
> 您好，我也十分关注bit-depth 的影响，能否给出原始图像数据转换为此存储库中提供的.png文件的处理脚本，十分感谢！

感谢关注，找了很久只找到了一个早期的版本，应该能用，将raw处理为png：
https://github.com/Linwei-Chen/LIS/tree/main/utils
供你参考

**Comment by LeonSakura** [comment]
十分感谢您的回复，请问raw转png的位深是通过这句代码中的output_bps控制的吗
out = raw.postprocess(use_camera_wb=False, user_wb=[1,1,1,1], half_size=False, no_auto_bright=True, output_bps=8, bright=1, user_black=None, user_sat=None)
但是在后续任务中，若用opencv读取保存的png，其读取的值好像还是0-255之间的值，这样是否会导致信息的损失呢，期待您的回复，感谢！

**Comment by Linwei-Chen** [comment]
是的，您可以参考rawpy的说明
https://stackoverflow.com/questions/57196366/meaning-of-output-bps-in-rawpy-params
